core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3681,4 +3681,58 @@ protected HumanReadableBytes.UnitSystem getUnitSystem()
       return HumanReadableBytes.UnitSystem.DECIMAL;
     }
   }
+
+
+  class DateExpandFunction implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "date_expand";
+    }
+
+    @Override
+    public void validateArguments(List<Expr> args)
+    {
+      validationHelperCheckArgumentCount(args, 3);
+    }
+
+    @Nullable
+    @Override
+    public ExpressionType getOutputType(Expr.InputBindingInspector inspector, List<Expr> args)
+    {
+      return ExpressionType.LONG_ARRAY;
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      final ExprEval expr = args.get(0).eval(bindings);
+      final long time = expr.asLong();
+
+      final int start = args.get(1).eval(bindings).asInt();
+      final int end = args.get(2).eval(bindings).asInt();
+
+
+      Long[] s = new Long[end - start + 1];
```

Comment:
most places have standardized array types to use `Object[]` for values instead of `Long[]`, so please use that here to be consistent

Reply:
@clintropolis - do you have a class that @findingrish can use as an example?

Reply:
what kind of example? everything looks fine otherwise, I believe the other array functions in this file should also be using `Object[]` for array value representation.